javagen/src/main/java/com/azure/autorest/template/ConvenienceAsyncMethodTemplate.java

```diff
@@ -143,28 +154,45 @@ private String expressionConvertFromBinaryData(IType responseBodyType, IType raw
         }
     }
 
-    private String expressionMapFromBinaryData(IType responseBodyType, IType rawType, Set<GenericType> typeReferenceStaticClasses) {
+    private String expressionMapFromBinaryData(IType responseBodyType, IType rawType,
+                                               Set<String> mediaTypes,
+                                               Set<GenericType> typeReferenceStaticClasses) {
         String mapExpression = null;
-        if (responseBodyType instanceof EnumType) {
-            // enum
-            mapExpression = String.format("protocolMethodData -> %1$s.from%2$s(protocolMethodData.toObject(%2$s.class))", responseBodyType, ((EnumType) responseBodyType).getElementType());
-        } else if (responseBodyType instanceof GenericType) {
-            // generic, e.g. list, map
-            typeReferenceStaticClasses.add((GenericType) responseBodyType);
-            mapExpression = String.format("protocolMethodData -> protocolMethodData.toObject(%1$s)", TemplateUtil.getTypeReferenceCreation(responseBodyType));
-        } else if (responseBodyType == ClassType.BINARY_DATA) {
-            // BinaryData, no need to do the map in expressionConvertFromBinaryData
-            mapExpression = null;
-        } else if (isModelOrBuiltin(responseBodyType)) {
-            // class
-            mapExpression = String.format("protocolMethodData -> protocolMethodData.toObject(%1$s.class)", responseBodyType.asNullable());
-        } else if (responseBodyType == ArrayType.BYTE_ARRAY) {
-            // byte[]
-            if (rawType == ClassType.BASE_64_URL) {
-                return "protocolMethodData -> protocolMethodData.toObject(Base64Url.class).decodedBytes()";
-            } else {
-                return "protocolMethodData -> protocolMethodData.toObject(byte[].class)";
-            }
+        SupportedMimeType mimeType = getResponseKnownMimeType(mediaTypes);
+        // TODO (weidxu): support XML etc.
```

Comment:
Currently I seems not able to get a XML type ObjectSerializer from azure-core, nor from azure-xml.